src/ParallelAlgorithms/Events/MonitorMemory.hpp

```diff
@@ -0,0 +1,351 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+#include <limits>
+#include <optional>
+#include <pup.h>
+#include <string>
+#include <type_traits>
+#include <unordered_map>
+#include <utility>
+#include <vector>
+
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "DataStructures/DataBox/TagName.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "Domain/Structure/ElementId.hpp"
+#include "Domain/Tags.hpp"
+#include "IO/Observer/Helpers.hpp"
+#include "IO/Observer/ObservationId.hpp"
+#include "IO/Observer/ObserverComponent.hpp"
+#include "IO/Observer/ReductionActions.hpp"
+#include "IO/Observer/Tags.hpp"
+#include "IO/Observer/TypeOfObservation.hpp"
+#include "Options/Auto.hpp"
+#include "Options/Options.hpp"
+#include "Parallel/CharmPupable.hpp"
+#include "Parallel/GlobalCache.hpp"
+#include "Parallel/Info.hpp"
+#include "Parallel/Invoke.hpp"
+#include "Parallel/Local.hpp"
+#include "Parallel/MemoryMonitor/MemoryMonitor.hpp"
+#include "Parallel/MemoryMonitor/Tags.hpp"
+#include "Parallel/Reduction.hpp"
+#include "Parallel/Serialize.hpp"
+#include "Parallel/TypeTraits.hpp"
+#include "ParallelAlgorithms/Actions/MemoryMonitor/ProcessArray.hpp"
+#include "ParallelAlgorithms/Actions/MemoryMonitor/ProcessGroups.hpp"
+#include "ParallelAlgorithms/Actions/MemoryMonitor/ProcessSingleton.hpp"
+#include "ParallelAlgorithms/EventsAndTriggers/Event.hpp"
+#include "Utilities/ErrorHandling/Error.hpp"
+#include "Utilities/Functional.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+namespace Parallel::Algorithms {
+struct Array;
+struct Group;
+struct Nodegroup;
+struct Singleton;
+}  // namespace Parallel::Algorithms
+/// \endcond
+
+namespace Events {
+/*!
+ * \brief Event run on the DgElementArray that will monitor the memory usage of
+ * parallel components.
+ *
+ * \details Given a list of parallel component names from Options, this will
+ * calculate the memory usage of each component and write it to disk in the
+ * reductions file under the `/MemoryMonitors/` group. The name of each file is
+ * the `pretty_type::name` of each parallel component.
+ *
+ * The parallel components available to monitor are the ones defined in the
+ * `component_list` type alias in the metavariables. In addition to these
+ * components, you can also monitor the size of the GlobalCache. Currently, you
+ * cannot monitor the size of the MutableGlobalCache. To see which parallel
+ * components are available to monitor, look in the metavariables for that
+ * executable. If you request to monitor an invalid parallel component, an ERROR
+ * will occur and a list of the available components to monitor will be printed.
```

Comment:
Reading the error message is probably more useful than reading the metavariables.  I'd suggest that first (and possibly only).